src/main/java/pro/sky/course2/employeebook/service/implementation/DepartmentServiceImpl.java

```diff
@@ -0,0 +1,49 @@
+package pro.sky.course2.employeebook.service.implementation;
+
+import org.springframework.stereotype.Service;
+import pro.sky.course2.employeebook.data.Employee;
+import pro.sky.course2.employeebook.service.DepartmentService;
+import pro.sky.course2.employeebook.service.EmployeeService;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class DepartmentServiceImpl implements DepartmentService {
+    private final EmployeeService employeeService;
+
+    DepartmentServiceImpl(EmployeeService employeeService) {
+        this.employeeService = employeeService;
+    }
+
+    @Override
+    public Employee findEmployeeWithMaxSalary(int departmentId) {
+        return employeeService.getMap().values().stream()
+                .filter(e -> e.getDepartment() == departmentId)
+                .max(Comparator.comparingDouble(Employee::getSalary))
+                .orElseThrow();
```

Comment:
мы должны передавать сюда функцию по созданию исключения
```suggestion
                .orElseThrow(() -> new IllegalArgumentException("Empty department"));
```